src/nodes/chunk_dispatch/chunk_dispatch.c

```diff
@@ -148,8 +147,17 @@ ts_chunk_dispatch_get_chunk_insert_state(ChunkDispatch *dispatch, Point *point,
 		ts_set_compression_status(cis, chunk);
 
 		ts_subspace_store_add(dispatch->cache, chunk->cube, cis, destroy_chunk_insert_state);
+	}
+	else if (cis->rel->rd_id == dispatch->prev_cis_oid && cis == dispatch->prev_cis)
+	{
+		/* got the same item from cache as before */
+		cis_changed = false;
+	}
 
-		if (found && ts_chunk_is_compressed(chunk) && !ts_chunk_is_distributed(chunk))
+	if (found)
+	{
+		Chunk *chunk = ts_chunk_get_by_id(cis->chunk_id, true);
```

Comment:
Could you add a comment explaining why you need to re-fetch the chunk using the chunk_id from the insert state. It is not very clear why this is necessary.

Reply:
the `decompress_batches_for_insert` needs it - so from now on every row inserted will need to have a `Chunk` struct.

the instructions above `ts_set_compression_status` is essentially bypasses the cache; and loads a new instance see [here](https://github.com/timescale/timescaledb/commit/84b6783a199c37bff6b36a3c0940915e5ca569ce#diff-7a9e9490252947c6a5c13cab7ce5ce5a2b6b67a5820c499b74b93f29a55977caR147) - so I'm not sure if we could trust the cache behind `ts_hypertable_find_chunk_for_point` ; or that will just cause trouble? (fyi: @antekresic)

for now I've decided to:
* lazy-init the `chunk` field
* load it with `ts_chunk_get_by_id` for now - in case its not already available

Reply:
Please add a comment describing the situation. It will help your future self understanding the rationale behind it.